src/users.spec.ts

```diff
@@ -235,16 +235,20 @@ describe("test user wallets", () => {
         assert.equal(transaction.getSignature().hex(), "c0bd2b3b33a07b9cc5ee7435228acb0936b3829c7008aacabceea35163e555e19a34def2c03a895cf36b0bcec30a7e11215c11efc0da29294a11234eb2b3b906");
     });
 
-    it("signs a general message", function () {
+    it("signs a general message", async function () {
         let signer = new UserSigner(UserSecretKey.fromString("1a927e2af5306a9bb2ea777f73e06ecc0ac9aaa72fb4ea3fecf659451394cccf"));
         let verifier = new UserVerifier(UserSecretKey.fromString("1a927e2af5306a9bb2ea777f73e06ecc0ac9aaa72fb4ea3fecf659451394cccf").generatePublicKey());
         const message = new TestMessage({
             foo: "hello",
             bar: "world"
         });
 
-        signer.sign(message);
+        await signer.sign(message);
+
+        const serialized = message.serializeForSigning();
+        const signature = Buffer.from(message.signature, "hex");
```

Comment:
Test will be improved in the very next PR (a secondary simplification is pending).